diff --git a/hw/xbox/mcpx/apu/vp/vp.c b/hw/xbox/mcpx/apu/vp/vp.c
index c57ecb634f..d4bae1836c 100644
--- a/hw/xbox/mcpx/apu/vp/vp.c
+++ b/hw/xbox/mcpx/apu/vp/vp.c
@@ -1768,7 +1768,10 @@ static void voice_work_init(MCPXAPUState *d)
 {
     VoiceWorkDispatch *vwd = &d->vp.voice_work_dispatch;
 
-    int num_workers = g_config.audio.vp.num_workers ?: SDL_GetNumLogicalCPUCores();
+    int num_workers = g_config.audio.vp.num_workers;
+    if (!num_workers) {
+        num_workers = MIN(SDL_GetNumLogicalCPUCores(), 4);
+    }
     vwd->num_workers = MAX(1, MIN(num_workers, MAX_VOICE_WORKERS));
     vwd->workers = g_malloc0_n(vwd->num_workers, sizeof(VoiceWorker));
     vwd->workers_should_exit = false;
